src/Functions/tokens.cpp

```diff
@@ -18,12 +18,75 @@ namespace ErrorCodes
     extern const int ILLEGAL_COLUMN;
 }
 
+namespace
+{
+template <typename T>
+std::optional<T> getArgument(const ColumnsWithTypeAndName & arguments, size_t argument_index, std::string_view function_name = "")
+{
+    static_assert(std::same_as<T, std::string_view> || std::same_as<T, UInt64> || std::same_as<T, std::vector<String>>);
+
+    if (argument_index >= arguments.size())
+        return {};
+
+    else if constexpr (std::same_as<T, std::string_view>)
+        return arguments[argument_index].column->getDataAt(0).toView();
+    else if constexpr (std::same_as<T, UInt64>)
+        return arguments[argument_index].column->getUInt(0);
+    else if constexpr (std::same_as<T, std::vector<String>>)
+    {
+        if (const ColumnConst * col_separators_const = checkAndGetColumnConst<ColumnArray>(arguments[argument_index].column.get()))
+        {
+            std::vector<String> values;
+            const Array & array = col_separators_const->getValue<Array>();
+            for (const auto & entry : array)
+                values.emplace_back(entry.safeGet<String>());
+            return values;
+        }
+        if (const ColumnArray * col_separators_non_const = checkAndGetColumn<ColumnArray>(arguments[argument_index].column.get()))
+        {
+            std::vector<String> values;
+            Field array_field;
+            col_separators_non_const->get(0, array_field);
+            Array & array = array_field.safeGet<Array>();
+            for (const auto & entry : array)
+                values.emplace_back(entry.safeGet<String>());
+            return values;
+        }
+        throw Exception(
+            ErrorCodes::ILLEGAL_COLUMN,
+            "Argument at index {} of function {} should be Array(String), got: {}",
+            argument_index + 1,
+            function_name,
+            arguments[argument_index].column->getFamilyName());
+    }
+}
+
+template <typename... Args>
+auto getArgumentAsStringArray(Args &&... args)
+{
+    return getArgument<std::vector<String>>(std::forward<Args>(args)...);
+}
+
+template <typename... Args>
+auto getArgumentAsStringView(Args &&... args)
+{
+    return getArgument<std::string_view>(std::forward<Args>(args)...);
+}
+
+template <typename... Args>
+auto getArgumentAsUInt(Args &&... args)
+{
+    return getArgument<UInt64>(std::forward<Args>(args)...);
+}
```

Comment:
As said before, if you want to specialize the `getArgument` with these 3 functions, a simple overload will work, not needed a template inside because there is not code overlap.

If you want a template any way, then 3 specializations with an undefined generic implementation (as suggested above) is better, but then these 3 wrappers are out of place.

Using a variadic template to wrap a fixed argument function with `std::forward` over-complicates things